shared/actions/chat/inbox.js

```diff
@@ -76,12 +76,14 @@ function* _backgroundUnboxLoop() {
 // Update inboxes that have been reset
 function* _updateFinalized(inbox: ChatTypes.GetInboxLocalRes) {
   const finalizedState: Constants.FinalizedState = Map(
-    (inbox.conversationsUnverified || []).reduce((map, convoUnverified) => {
-      return map.set(
-        Constants.conversationIDToKey(convoUnverified.metadata.conversationID),
-        convoUnverified.metadata.finalizeInfo
-      )
-    }, Map())
+    (inbox.conversationsUnverified || [])
+      .filter(c => c.metadata.finalizeInfo)
+      .reduce((map, convoUnverified) => {
```

Comment:
This will be faster if you don't reduce into the same map.

It's generally faster to make the values and then put them in a map. So this can be rewritten as:

```js
Map(
  ...
    .filter(/*...*/)
    .map(convoUnverified => 
      [Constants.conversationIDToKey(convoUnverified.metadata.conversationID),
       convoUnverified.metadata.finalizeInfo]
     )
)
```